### PR TITLE
mobile: retry starting emulator up to 3 times

### PR DESCRIPTION
**Pull request by jpsim**
Commit Message:
Additional Description:
Risk Level: Low, CI only
Testing: CI
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: .github/workflows/android_build.yml [hunk 1]
```diff
@@ -56,9 +56,13 @@ jobs:
     - run: cd mobile && ./ci/mac_ci_setup.sh --android
       if: steps.should_run.outputs.run_ci_job == 'true'
       name: 'Install dependencies'
-    - name: 'Start simulator'
+    - uses: nick-fields/retry@943e742917ac94714d2f408a0e8320f2d1fcafcd
       if: steps.should_run.outputs.run_ci_job == 'true'
-      run: cd mobile && ./ci/mac_start_emulator.sh
+      name: 'Start emulator'
+      with:
+        timeout_minutes: 10
```
Comment (Augustyniak) [inline .github/workflows/android_build.yml:63]: could we lower it? Not sure whether 5 is not too low but I think that we should be fine with something that's lower than 10. Maybe somewhere between 5 and 7?

Comment (jpsim) [reply .github/workflows/android_build.yml:63]: Took 4:41s here: https://github.com/envoyproxy/envoy/actions/runs/3905297829/jobs/6673475684

Let's try with 6.

File: .github/workflows/android_build.yml [hunk 2]
```diff
@@ -97,9 +101,13 @@ jobs:
     - name: 'Install dependencies'
       if: steps.should_run.outputs.run_ci_job == 'true'
       run: cd mobile && ./ci/mac_ci_setup.sh --android
-    - name: 'Start simulator'
+    - uses: nick-fields/retry@943e742917ac94714d2f408a0e8320f2d1fcafcd
       if: steps.should_run.outputs.run_ci_job == 'true'
-      run: cd mobile && ./ci/mac_start_emulator.sh
+      name: 'Start simulator'
```
Comment (RyanTheOptimist) [inline .github/workflows/android_build.yml:106]: If it's not too much trouble, mind changing this from "simulator" to "emulator" like you did on line 61? (Here and the two others below)

**Review by RyanTheOptimist — COMMENTED** [review]
Oh, this is very cool! Fingers crossed.

File: .github/workflows/android_build.yml (hunk 2 above)
Comment (jpsim) [reply .github/workflows/android_build.yml:106]: Oops, thanks for catching the ones I missed.

**Review by Augustyniak — APPROVED** [review]
Fingers crossed for this helping us.

**Pull request merged**